engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -97,127 +98,122 @@ void postConstruct() {
-            LOGGER.info("Error while processing aggregation", t);
+        try {
+            statelessSessionFactory.withSession(statelessSession -> {
+                processAggregateEmailsByAggregationKey(
+                        aggregationCommand.getAggregationKey(),
+                        aggregationCommand.getStart(),
+                        aggregationCommand.getEnd(),
+                        aggregationCommand.getSubscriptionType(),
+                        // Delete on daily
+                        aggregationCommand.getSubscriptionType().equals(EmailSubscriptionType.DAILY));
+            });
+        } catch (Exception e) {
+            LOGGER.info("Error while processing aggregation", e);
             failedAggregationCommandCount.increment();
-            return Uni.createFrom().nullItem();
-        });
+        }
     }
 
-    private Multi<Tuple2<NotificationHistory, EmailAggregationKey>> processAggregateEmailsByAggregationKey(EmailAggregationKey aggregationKey, LocalDateTime startTime, LocalDateTime endTime, EmailSubscriptionType emailSubscriptionType, boolean delete) {
+    private List<Tuple2<NotificationHistory, EmailAggregationKey>> processAggregateEmailsByAggregationKey(EmailAggregationKey aggregationKey, LocalDateTime startTime, LocalDateTime endTime, EmailSubscriptionType emailSubscriptionType, boolean delete) {
         final EmailTemplate emailTemplate = emailTemplateFactory.get(aggregationKey.getBundle(), aggregationKey.getApplication());
 
-        Multi<Tuple2<NotificationHistory, EmailAggregationKey>> doDelete = delete ?
-                emailAggregationRepository.purgeOldAggregation(aggregationKey, endTime)
-                        .onItem().transformToMulti(unused -> Multi.createFrom().empty()) :
-                Multi.createFrom().empty();
+        if (delete) {
+            // TODO NOTIF-488 To be changed
+            emailAggregationRepository.purgeOldAggregation(aggregationKey, endTime);
+        }
 
         if (!emailTemplate.isEmailSubscriptionSupported(emailSubscriptionType)) {
-            return doDelete;
+            return Collections.emptyList();
         }
 
         TemplateInstance subject = emailTemplate.getTitle(null, emailSubscriptionType);
         TemplateInstance body = emailTemplate.getBody(null, emailSubscriptionType);
 
         if (subject == null || body == null) {
-            return doDelete;
+            return Collections.emptyList();
         }
-        return emailAggregator.getAggregated(aggregationKey, emailSubscriptionType, startTime, endTime)
-                .onItem().transform(Map::entrySet)
-                .onItem().transformToMulti(Multi.createFrom()::iterable)
-                .onItem().transformToMultiAndConcatenate(entries -> {
+
+        Set<Map.Entry<User, Map<String, Object>>> bla = emailAggregator.getAggregated(aggregationKey, emailSubscriptionType, startTime, endTime)
+                .entrySet();
+
+        return bla.stream()
```

Comment:
![image](https://user-images.githubusercontent.com/3845764/157118712-432070d6-5d52-46c8-9554-b147dd50f143.png)

Reply:
hahaha :-D

Reply:
That wasn't meant to be pushed 😄 Removed.